database/migrations/2021_12_12_171955_create_product_warning_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateProductWarningTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('product_warning', function (Blueprint $table) {
+            $table->foreignId('product_id')
+                ->references('id')->on('products')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+            $table->bigInteger('warning_id')->unsigned()->nullable(false);
+            $table->foreign('warning_id')
+                ->references('id')->on('warnings')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+            $table->unique(['product_id', 'warning_id']);
```

Comment:
instead primary key could be used. But it is not mandatory.

Reply:
The main idea is to not be able to insert product 'A' with warning 'B' multiple times.